reflex/.templates/web/utils/state.js

```diff
@@ -4,6 +4,8 @@ import io from "socket.io-client";
 import JSON5 from "json5";
 import env from "env.json";
 import Cookies from "universal-cookie";
+import { useEffect, useReducer, useRef, useState } from "react";
+import Router from "next/router";
```

Comment:
i was having a problem with a stale `router` reference being in the websocket event handler closures. i tried to fix it by making `router` a `useRef`, like `socket`, but this wasn't really working either because the ref was never getting set properly and seemed messier (more coupling between page and state). i can take another crack at it though... although it seemed to be working with the global router; maybe i can call `useRouter` from inside `useEventLoop` and side-step the coupling problem.